torch/nn/utils/clip_grad.py

```diff
@@ -6,7 +6,9 @@
 _tensor_or_tensors = Union[torch.Tensor, Iterable[torch.Tensor]]
 
 
-def clip_grad_norm_(parameters: _tensor_or_tensors, max_norm: float, norm_type: float = 2.0) -> torch.Tensor:
+def clip_grad_norm_(
```

Comment:
why do we have separate python and c++ implementations of clip_grad_norm, and should we be calling c++ implementation from here? 
cc @ailzhang @JackCaoG because  XLA used to rely on this (patched) python implementation - is it possible to remove this dependency.

Reply:
Yeah I was wondering the same question above. Definitely seems better to merge them

Reply:
I had an impression that I benchmarked the two implementations on a pytorch/xla model on TPU and saw some non-trivial perf gap but couldn't recall the details. But yea let's put this on the issue https://github.com/pytorch/xla/issues/2824 and track it properly. 
To unblock this PR for now, we can just change the signature in an xla PR and land it after this one.